File: dbflow-sqlcipher/src/main/java/com/raizlabs/dbflow/sqlcipher/SQLCipherStatement.java
package com.raizlabs.dbflow.sqlcipher;

import com.raizlabs.android.dbflow.structure.database.DatabaseStatement;

import net.sqlcipher.database.SQLiteStatement;

/**
 * Description:
 */
public class SQLCipherStatement implements DatabaseStatement {

    private final SQLiteStatement statement;

    public static SQLCipherStatement from(SQLiteStatement statement) {
        return new SQLCipherStatement(statement);
    }

    SQLCipherStatement(SQLiteStatement statement) {
        this.statement = statement;
    }

    @Override
    public long executeUpdateDelete() {
        return statement.executeUpdateDelete();
    }

    @Override
    public void execute() {
        statement.execute();
    }

    @Override
    public void close() {
        statement.close();
    }

    @Override
    public long simpleQueryForLong() {
        return statement.simpleQueryForLong();
    }

    @Override
    public String simpleQueryForString() {
        return statement.simpleQueryForString();
    }

    @Override
    public long executeInsert() {
        return statement.executeInsert();
    }

    @Override
    public void bindString(int index, String name) {
        statement.bindString(index, name);
    }

    @Override
    public void bindNull(int index) {
        statement.bindNull(index);
    }

    @Override
    public void bindLong(int index, long aLong) {
        statement.bindLong(index, aLong);
    }

    @Override
    public void bindDouble(int index, double aDouble) {
        statement.bindDouble(index, aDouble);
    }

    @Override
    public void bindBlob(int index, byte[] bytes) {
        statement.bindBlob(index, bytes);
    }
}

File: dbflow-sqlcipher/src/main/java/com/raizlabs/dbflow5/sqlcipher/SQLCipherStatement.kt
package com.raizlabs.dbflow5.sqlcipher

import com.raizlabs.dbflow5.database.BaseDatabaseStatement
import com.raizlabs.dbflow5.database.DatabaseStatement

import net.sqlcipher.database.SQLiteStatement

/**
 * Description: Implements the methods necessary for [DatabaseStatement]. Delegates calls to
 * the contained [SQLiteStatement].
 */
class SQLCipherStatement
internal constructor(val statement: SQLiteStatement) : BaseDatabaseStatement() {

    override fun executeUpdateDelete(): Long = statement.executeUpdateDelete().toLong()

    override fun execute() {
        statement.execute()
    }

    override fun close() {
        statement.close()
    }

    override fun simpleQueryForLong(): Long {
        return statement.simpleQueryForLong()
    }

    override fun simpleQueryForString(): String? {
        return statement.simpleQueryForString()
    }

    override fun executeInsert(): Long {
        return statement.executeInsert()
    }

    override fun bindString(index: Int, s: String) {
        statement.bindString(index, s)
    }

    override fun bindNull(index: Int) {
        statement.bindNull(index)
    }

    override fun bindLong(index: Int, aLong: Long) {
        statement.bindLong(index, aLong)
    }

    override fun bindDouble(index: Int, aDouble: Double) {
        statement.bindDouble(index, aDouble)
    }

    override fun bindBlob(index: Int, bytes: ByteArray) {
        statement.bindBlob(index, bytes)
    }

    companion object {

        @JvmStatic
        fun from(statement: SQLiteStatement): SQLCipherStatement = SQLCipherStatement(statement)
    }
}
